src/cudamatrix/cu-matrix.cc

```diff
@@ -2528,8 +2528,45 @@ void CuMatrixBase<Real>::AddElements(Real alpha,
 }
 
 template<typename Real>
+void CuMatrixBase<Real>::AddElements(Real alpha, const CuArray<Int32Pair> &indexes,
+				     const Real *input) {
+  KALDI_ASSERT(input != NULL);
+  // Checks the dimension.
+  MatrixIndexT num_rows = this->num_rows_, num_cols = this->num_cols_;
+  std::vector<Int32Pair> cpu_indexes;
+  indexes.CopyToVec(&cpu_indexes);
+  for (int32 i = 0; i < indexes.Dim(); ++i) {
```

Comment:
Please don't copy the indexes to CPU just for the sake of a check.  The whole point of giving a CuArray as argument to the function was to avoid this type of operation.